action.go

```diff
@@ -75,10 +74,6 @@ type BaseAction struct {
 	Description string
 }
 
-func (b *BaseAction) LogStart() {
```

Comment:
to remove duplicates of Printf, I think it'd be best to keep this function and call it from the the main function in the two places ?

Reply:
Not sure I follow: are you saying that using two "log.Printf(...)" lines is too much repetition?

In general it make sense to keep some form of abstraction assuming a) it's more than one-liner b) a specific format is used and/or c) there are symmetric LogEnd/LogFoo helpers. Without any of these the function seems like a unnecessary distraction IMHO.

Reply:
Yeah I think two separate instances is too much repetition for something like this where the same behaviour is in two places - if the style of one printf call is modified, the other call (just for nested recipes) may get missed. I'd prefer to move this function to `cmd/main/debos.go`.

Reply:
Shouldn't there be a CI test to catch such changes/issues? Would it make sense to add "LogEnd" in that case?

Reply:
My opinion is no to both of those questions.

Reply:
Your suggestion does not seem to be doable. For example, applying the following simple diff fails to build as below.

```diff
index 9f390db..0c2e426 100644
--- a/actions/pacman_action.go
+++ b/actions/pacman_action.go
@@ -25,8 +25,6 @@ type PacmanAction struct {
 }

 func (p *PacmanAction) Run(context *debos.DebosContext) error {
-       p.LogStart()
-
        pacmanOptions := []string{"pacman", "-Syu", "--noconfirm"}
        pacmanOptions = append(pacmanOptions, p.Packages...)

diff --git a/actions/pacstrap_action.go b/actions/pacstrap_action.go
index 25c1624..e0eb9d2 100644
--- a/actions/pacstrap_action.go
+++ b/actions/pacstrap_action.go
@@ -84,8 +84,6 @@ func (d *PacstrapAction) PreMachine(context *debos.DebosContext, m *fakemachine.
 }

 func (d *PacstrapAction) Run(context *debos.DebosContext) error {
-       d.LogStart()
-
        files := map[string]string{
                "/etc/pacman.conf":         d.Config,
                "/etc/pacman.d/mirrorlist": d.Mirror,
diff --git a/cmd/debos/debos.go b/cmd/debos/debos.go
index 951cd2e..513b3b7 100644
--- a/cmd/debos/debos.go
+++ b/cmd/debos/debos.go
@@ -15,6 +15,10 @@ import (
        "github.com/jessevdk/go-flags"
 )

+func (b *debos.BaseAction) LogStart() {
+       log.Printf("==== %s ====\n", b)
+}
+
 func checkError(context *debos.DebosContext, err error, a debos.Action, stage string) int {
        if err == nil {
                return 0
```

```shell
$ go build -v ./...
github.com/go-debos/debos/cmd/debos
# github.com/go-debos/debos/cmd/debos
cmd/debos/debos.go:18:7: cannot define new methods on non-local type debos.BaseAction
````

Reply:
> Your suggestion does not seem to be doable. For example, applying the following simple diff fails to build as below.

Maybe you missed my edit to the suggestion:
> I'd prefer to keep this function and call it from `cmd/main/debos.go` in the two places where it is used.

Reply:
@evelikov I implemented the above suggestion at https://github.com/go-debos/debos/pull/417 and will close this PR. Can you review it?